lib/ghompcgs.gi

```diff
@@ -353,25 +353,38 @@ InversePcgs := function( hom )
       
         # add it
         hom!.rangePcgs := InducedPcgsByPcSequenceNC( pcgs, gensInv ); 
-        hom!.rangePcgsPreimages := imgsInv;
-#T better MakeImmutable
+        hom!.rangePcgsPreimages := Immutable(imgsInv);
         
-        # we have the kernel also
-        SetKernelOfMultiplicativeGeneralMapping( hom, SubgroupNC(Source(hom),
-                                                          gensKer ) );
-  
+        # we have the kernel also, if needed (or we check).
+        if not HasKernelOfMultiplicativeGeneralMapping(hom)
+          or InfoLevel(InfoAttributes)>1 then
+          #Check whether the Pcgs is for the whole group.
+          # Otherwise there is a kernel that will not be visible in
+          # the modulo pcgs that the homomorphism uses
+          tmp:=DenominatorOfModuloPcgs(hom!.sourcePcgs);
```

Comment:
Just for my understanding: This is "only" necessary because `TryPcgsPermGroup` does not invoke `SetDenominatorOfModuloPcgs`, right? If so, I guess we could also fix it with a patch like this? (To clarify, I am not requesting a change to this PR, I merely want to check my understanding)

```patch
diff --git a/lib/pcgsperm.gi b/lib/pcgsperm.gi
index 414ce53eb..2f8c0914b 100644
--- a/lib/pcgsperm.gi
+++ b/lib/pcgsperm.gi
@@ -432,6 +432,10 @@ InstallGlobalFunction(TryPcgsPermGroup,function(arg)
             SetIsNilpotentGroup( grp, true );
         fi;
     else
+        U := G[ Length( G ) ];
+        # set numerator and denominator appropriately
+        SetNumeratorOfModuloPcgs(pcgs,GeneratorsOfGroup(G));
+        SetDenominatorOfModuloPcgs(pcgs,GeneratorsOfGroup(U));
         pcgs!.denominator := G[ Length( G ) ];
         if     HasIsSolvableGroup( G[ Length( G ) ] )
            and IsSolvableGroup( G[ Length( G ) ] )  then
```

Reply:
In principle yes. The issue is with the original definition (that was documented): DenominatorOfModuloPcgs is supposed to be a Pcgs (or fail, if it is not a pcgs), not a generating set of a nonsolvable group. So the denominator information for a solvable factor group of an insolvable group must be stored differently.